### ISSUE TITLE
Acri being scaled weirdly when display scaling isn't 100%

### ISSUE THREAD

**Issue by Lordmau5** [obsproject/obs-studio#4058]
## Platform

Operating system and version: Windows 10 20H2
OBS Studio version: 26.1.1

## Expected Behavior

The Acri theme should be scaled properly like in 26.0.2

## Current Behavior

The Acri theme is scaling way too big on display scaling that is higher than 100%

## Steps to Reproduce

1. Set a monitor to display scaling above 100% (e.g. I have a 4K monitor that I have set to 150%)
2. Move OBS to that monitor
3. Select the Acri theme
4. It should be scaled just like in 26.0.2 (or like the other themes in 26.1.1 since they seem to work just fine)

## Additional information

I've heard that due to an update in the Qt engine the scaling got fixed in 26.1.0 of OBS.
However, it seems that Acri has suffered some issues from this.

I've checked with other themes and they seem to scale perfectly fine.

**OBS 26.0.2 and 26.1.1 with the Acri theme on 150% display scaling side-by-side**
![1610495372-39305](https://user-images.githubusercontent.com/1345036/104391950-54ddf680-5541-11eb-8f21-f284f6f440c2.jpg)

**OBS 26.0.2 and 26.1.1 with the Default (Dark) theme on 150% display scaling side-by-side**
![1610495446-39306](https://user-images.githubusercontent.com/1345036/104391953-56a7ba00-5541-11eb-81ca-06c448e5388c.jpg)

As can be seen on the section names they are slightly cut off due to the scaling.
It's even worse with 300% scaling (though I personally would never want to run at this setting)
![image](https://user-images.githubusercontent.com/1345036/104392152-ccac2100-5541-11eb-94be-46f1a4fb51a4.png)

I wonder if it'd be possible to have an option in the theme to turn off this scaling?
Again, it seems far better with the Standard (Dark) theme on 300% scaling
![image](https://user-images.githubusercontent.com/1345036/104392226-f107fd80-5541-11eb-8bb8-73757ff7a38a.png)

**Comment by WizardCM** [comment]
To confirm my theory, I grabbed your screenshot, scaled the top left 26.0.2 example to 150% in PhotoShop, and placed it above the 150% example in the same screenshot. The sizing matches correctly (near-pixel-perfect):

![image](https://user-images.githubusercontent.com/941350/104393307-3fe08200-5598-11eb-9c19-13e808e6bf85.png)

I then did the same with your 300% example:

![image](https://user-images.githubusercontent.com/941350/104393457-9e0d6500-5598-11eb-99b8-1c5a94d4ae3a.png)


The only bug I see is that the dock headings don't properly handle larger font sizes in scaling like 300% so the text is cut off. The rest is by design, you're just used to everything being tiny.

**Comment by Lordmau5** [comment]
Yeah but again, why is it not happening on any of the other themes?

Only Acri has this scaling "fix" / these issues.

It was a perfect setup on 26.0.2 and it's the reason why I'm still stuck on that version.

If I'd upgrade to 26.1.x I'd not be using Acri for the forseeable future, if at all, due to this :( 

"Never change a running system" kinda thing... I can understand if you guys wouldn't want to bring back the old, "broken" scaling for Acri, but I am not the only one wishing for that (a friend of me is also not updating to 26.1 yet due to this)

**Comment by WizardCM** [comment]
> Yeah but again, why is it not happening on any of the other themes?

Which part isn't happening in the other themes? Incorrect height/font on dock headings? Probably caused by this line

https://github.com/obsproject/obs-studio/blob/64164a1523e616bb1168a3f61f3738b3a9ac3058/UI/data/themes/Acri.qss#L175

Warchamp struggled with padding a lot when working on Acri, and it wouldn't surprise me if it's the cause of unexpected behaviour when using DPI scaling.

Essentially, DPI scaling didn't work at all when using fractional scaling, so 125% and 150% would round to 100%, and 175% would round to 200%.

> If I'd upgrade to 26.1.x I'd not be using Acri for the forseeable future, if at all, due to this :(

I am confused by this statement. Why exactly?

**Comment by Lordmau5** [comment]
Okay, so I feel it didn't come over properly in the screenshots so I went ahead and recorded how it looks like from my phone.

As in, how I see it when looking at the monitor.

The TL;DR is that with the broken scaling / rounding on 26.0.2 the size of OBS was perfect for my use-case.
However, with it now scaling to 150% it's breaking my layout and I'd like there to be an option to force a scaling for OBS in the app itself (if that's doable with Qt?)

Video showing and explaining the issues in 26.1:
https://www.youtube.com/watch?v=0r0ddV0EQOQ

**Comment by WizardCM** [comment]
I don't personally expect us to add a scaling override, but I'll leave this open for others to contribute their thoughts.

For the moment, the workaround would be to set `QT_SCALE_FACTOR` to `1`:

Either via a launch script (.bat file)
```
@echo off
set QT_SCALE_FACTOR=1
start obs64.exe
```

Or via a system-wide Environment variable. Note that this would affect all apps that use Qt, and may require a system reboot to take effect.

This can be accessed via Advanced System Settings.

**Comment by WizardCM** [comment]
OK, I can't reproduce the Acri heading issue on Windows 10. :/

**Comment by Lordmau5** [comment]
> I don't personally expect us to add a scaling override, but I'll leave this open for others to contribute their thoughts.
> 
> For the moment, the workaround would be to set `QT_SCALE_FACTOR` to `1`:
> 
> Either via a launch script (.bat file)
> 
> ```
> @echo off
> set QT_SCALE_FACTOR=1
> start obs64.exe
> ```
> 
> Or via a system-wide Environment variable. Note that this would affect all apps that use Qt, and may require a system reboot to take effect. This can be accessed via Advanced System Settings.
> 
> Edit: As a side note, Discord does contain a scaling setting in Settings -> Appearance if you want to set that one to 150% and the rest of your monitor to 100%.

I've tried both a batch script and an environment variable (yes, even with a reboot afterwards), yet neither of them show OBS 26.1 as it looked like before.

Also, Discord in 150% would work from a readability point of view, but since the display is at 100% scaling, I have to drag my mouse cursor way further than I would if I were on 150% scaling.

Would a launch parameter for OBS be an option?
Basically something like `--force-dpi-scaling=1` or `--disable-high-dpi-scaling` or something?

**Comment by Lordmau5** [comment]
Are there any updates on this?

**Comment by RytoEX** [comment]
> ```
> @echo off
> set QT_SCALE_FACTOR=1
> start obs64.exe
> ```

If you're trying to turn of scaling entirely, try QT_ENABLE_HIGHDPI_SCALING or QT_AUTO_SCREEN_SCALE_FACTOR:
> The QT_ENABLE_HIGHDPI_SCALING environment variable, introduced in Qt 5.14, enables automatic scaling based on the pixel density of the monitor. Replaces QT_AUTO_SCREEN_SCALE_FACTOR.

https://doc.qt.io/qt-5/highdpi.html

**Comment by Lordmau5** [comment]
Closing this as it was merged with my PR in [this commit](https://github.com/obsproject/obs-studio/commit/5b899788dacb30448f12d1b86c67c5a27dab50c1) :)
